napari/layers/base/base.py

```diff
@@ -698,12 +700,14 @@ def _get_extent_world(self, data_extent):
     @property
     def extent(self) -> Extent:
         """Extent of layer in data and world coordinates."""
-        data = self._extent_data
-        return Extent(
-            data=data,
-            world=self._get_extent_world(data),
-            step=abs(self._data_to_world.scale),
-        )
+        extent_data = self._extent_data
+        if self._extent_cache is None:
+            self._extent_cache = Extent(
+                data=extent_data,
```

Comment:
[My comment in the previous PR](https://github.com/napari/napari/pull/1950#discussion_r747698274) was actually related to storing the result of `self._data_to_world`, not `self._extent_data`, but I guess we could store both?

Reply:
ohh i got you now - yup read that too quickly, have fixed